### ISSUE TITLE
Could not find the required redirect URL for your account 

### ISSUE THREAD

**Issue by skerit** [tootsuite/mastodon#2025]
External follows don't seem to work on my server.

I've seen a few other issues describing the same problem, but those involved problems with the .well-know path.

The .well-know path seems to work just fine on my server:

https://mastodon.develry.be/.well-known/host-meta

When I look at the log file, the .well-known requests reach mastodon, but it seems to respond to the `webfinger` request with a 404:

```
Apr 17 20:50:42 method=GET path=/.well-known/host-meta format=xml controller=WellKnown::HostMetaController action=show status=200 duration=3.16 view=1.06 db=0.00
Apr 17 20:50:42 method=GET path=/.well-known/webfinger format=json controller=WellKnown::WebfingerController action=show status=404 duration=1.38 view=0.00 db=0.00
```

* * * *

- [true] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [true] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

**Comment by mjankowski** [comment]
Possibly related to https://github.com/tootsuite/mastodon/pull/2008

**Comment by skerit** [comment]
Ah no, I found the problem.

I'm using a different `WEB_DOMAIN` (mastodon.develry.be) and `LOCAL_DOMAIN` (develry.be).

Once I created a redirect from `https://develry.be/.well-known/host-meta` to `https://mastodon.develry.be/.well-known/host-meta` things started working.

By the way: I did not create the original follow request using `skerit@develry.be` but `skerit@mastodon.develry.be`. While those `.well-known` requests worked, I guess mastodon thought the `@mastodon.develry.be` account didn't exist because it didn't match the `LOCAL_DOMAIN`?
